worker-core/src/main/java/com/hpe/caf/worker/core/StreamingWorkerWrapper.java

```diff
@@ -73,6 +76,7 @@ public void run()
             LOG.warn("Worker threw unhandled exception", e);
             workerTask.setResponse(worker.getGeneralFailureResult(e));
         }
+        MDC.remove(CORRELATION_ID);
```

Comment:
Put this in a `finally` block to make sure that it is done (even if one of the exception handlers throws an exception).

Reply:
Done